tests/jballerina-unit-test/src/test/resources/test-src/expressions/let/let-expression-test.bal

```diff
@@ -316,6 +316,15 @@ function getRecordConstrainedError() returns FooError {
     return e;
 }
 
+public function testAnonymousRecordWithLetExpression() {
```

Comment:
As discussed in https://github.com/ballerina-platform/ballerina-lang/issues/26906, since this test may no longer cover the issue we are fixing here, shall we add another test that does?

Reply:
Added 
```
public function testAnonymousRecordWithLetExpression() {
    record {
        int i;
        int j?;
    } rec = let int v = 1 in {i: v};
    
    assert(rec.i, 1);
    
    rec.j = 5;
    assert(rec?.j, 5);
}
```
to cover the scenario you had described in #26906 which relies on context information rather than assignability.